app/src/main/java/com/nowak/wjw/simplecompass/ui/main/MainFragment.java

```diff
@@ -71,54 +54,26 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         mBinding.setLifecycleOwner(getViewLifecycleOwner());
 
         updateValuesFromBundle(savedInstanceState);
-        Timber.d("isRequestingLocationUpdates  :%s", isRequestingLocationUpdates);
-
-        AppContainer appContainer = ((MyApplication) requireActivity().getApplication()).appContainer;
-        mViewModel = new ViewModelProvider(this).get(MainViewModel.class);
+        MainViewModelFactory mainViewModelFactory = ((MyApplication) getActivity().getApplication()).appContainer.mainViewModelFactory();
+        mViewModel = new ViewModelProvider(this, mainViewModelFactory).get(MainViewModel.class);
         mBinding.setViewModel(mViewModel);
-        mViewModel.getmFoundLastLocation().observe(getViewLifecycleOwner(), is -> {
-            if (!is.getHasBeenHandled()) {
-                Timber.d("vm.foundLocation: %s", is.peekContent());
-                foundLastLocation = is.getContentIfNotHandled();
+        mViewModel.needScreenOrientation.observe(getViewLifecycleOwner(), isNeeded -> {
+            if (isNeeded) {
+                int mScreenRotation = requireActivity().getWindowManager().getDefaultDisplay().getRotation();
+                mViewModel.provideScreenRotation(mScreenRotation);
```

Comment:
what would be a better way to do this?  
putting it inside onResume didn't work cause when you flip your phone by 180degrees there is no onCreate triggered.